packages/fether-react/src/Send/TxForm/TxDetails/TxDetails.js

```diff
@@ -8,36 +8,38 @@ import BigNumber from 'bignumber.js';
 import { fromWei, toWei } from '@parity/api/lib/util/wei';
 
 class TxDetails extends Component {
+  renderDetails = () => {
+    const { estimatedTxFee } = this.props;
+
+    return estimatedTxFee
+      ? `${this.renderCalculation()}
+${this.renderFee()}
+${this.renderTotalAmount()}`
+      : `
+Transaction incomplete..`;
+  };
+
   renderCalculation = () => {
     const { estimatedTxFee, values } = this.props;
-
     const gasPriceBn = new BigNumber(values.gasPrice.toString());
     const gasLimitBn = estimatedTxFee
       .div(gasPriceBn)
       .div(10 ** 9)
       .toFixed(0)
       .toString();
 
-    return `Estimate amount of gas: ${gasLimitBn}`;
-  };
-
-  renderDetails = () => {
-    return `${this.renderCalculation()}
-${this.renderFee()}
-${this.renderTotalAmount()}`;
+    return `Gas: ${gasLimitBn}`;
   };
 
   renderFee = () => {
     const { estimatedTxFee } = this.props;
-
     return `Fee: ${fromWei(estimatedTxFee, 'ether')
```

Comment:
I think for consistent styling we should retain the blank line before this `return`